### PR TITLE
New Recipe: CPUInfo v0.0.20200122

### PR DESCRIPTION
**Pull request by stemann**
Packaging similar to [libcpuinfo0](https://packages.ubuntu.com/impish/libcpuinfo0) in Ubuntu.

Starting with an old version - used by PyTorch v1.5.

### CODE REVIEW

File: C/CPUInfo/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    ..
+cmake --build . -- -j $nproc
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+filter!(p -> arch(p) != "aarch64" || os(p) != "macos", platforms) # aarch64-macos unsupported
+filter!(p -> os(p) != "freebsd", platforms) # FreeBSD unsupported
```
Comment (giordano) [inline C/CPUInfo/build_tarballs.jl:42]: Bit simpler
```suggestion
filter!(p -> !(Sys.isapple(p) && arch(p) == "aarch64"), platforms) # aarch64-macos unsupported
filter!(p -> !Sys.isfreebsd(p), platforms) # FreeBSD unsupported
```

**Comment by stemann** [comment]
@giordano Any idea why the Windows x86_64 build outputs a `.dll.a` "library"? I tried just naively renaming it, but it's not the proper object file type (COFF?).

**Comment by giordano** [comment]
The `*.dll.a` isn't a dynamic library but an [import library](https://en.wikipedia.org/wiki/Dynamic-link_library#Import_libraries).

**Comment by giordano** [comment]
Judging by the log, the library is compiled https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=17888&view=logs&j=e919827e-2bea-5cb6-e233-5480b9308c51&t=b8c4887f-47c5-564a-0700-a9fc4ceb14a8&l=237 but not installed?

File: C/CPUInfo/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    -DCMAKE_INSTALL_LIBDIR=$libdir \
+    -DCPUINFO_BUILD_UNIT_TESTS=OFF \
+    -DCPUINFO_BUILD_MOCK_TESTS=OFF \
+    -DCPUINFO_BUILD_BENCHMARKS=OFF \
+    -DCPUINFO_LIBRARY_TYPE=shared \
+    ..
+cmake --build . -- -j $nproc
+make install
+if [[ $target == *-w64-mingw32 ]]; then
+    cp -a libcpuinfo.dll $libdir
```
Comment (giordano) [inline C/CPUInfo/build_tarballs.jl:32]: Never use `cp` in this way
```suggestion
    install -Dvm 755 libcpuinfo.dll "${libdir}/libcpuinfo.dll"
```

**Comment by stemann** [comment]
Any tips for getting aarch64-macos going?
https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=17881&view=logs&j=3ec2715d-2957-5178-a168-848455b259d3&t=3b32053d-f6dd-54d4-cb8c-75ff43e1571f&l=268

**Comment by stemann** [comment]
> Judging by the log, the library is compiled https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=17888&view=logs&j=e919827e-2bea-5cb6-e233-5480b9308c51&t=b8c4887f-47c5-564a-0700-a9fc4ceb14a8&l=248 but not installed?

And indeed, it was there. Silly Windows builds... :-)

**Comment by giordano** [comment]
> Any tips for getting aarch64-macos going?

No

**Comment by stemann** [comment]
Will enable i686-windows and x86_64-freebsd in a later version.

**Comment by giordano** [comment]
> Will enable i686-windows and x86_64-freebsd in a later version.

I'm confused, why?  Didn't they work in https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=17894&view=results?

**Comment by stemann** [comment]
> > Will enable i686-windows and x86_64-freebsd in a later version.
> 
> I'm confused, why? Didn't they work in https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=17894&view=results?

The overall aim is to provide dependencies for Torch, notably XNNPACK which takes a long time to build - for #4554.

I thought I would start with something simple CPUInfo - and a sufficiently old version that I would not be missing some random version for all the PyTorch projects that depend on CPU Info.

File: C/CPUInfo/bundled/patches/lowercase-windows-include.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/src/x86/windows/init.c b/src/x86/windows/init.c
```
Comment (giordano) [inline C/CPUInfo/bundled/patches/lowercase-windows-include.patch:1]: It'd be good to upstream this patch

Comment (stemann) [reply C/CPUInfo/bundled/patches/lowercase-windows-include.patch:1]: They actually already have fixed that - luckily :-)

**Pull request merged**